### PR TITLE
docstrings: INPUT formatting

### PR DESCRIPTION
**Pull request by gmou3**
Separate ` ``x, y`` ` to ` ``x``, ``y`` ` and remove excessive newlines.

### CODE REVIEW

**Comment by kwankyu** [comment]
I recommend you to split this PR. Due to the huge number of files that this PR touches, the github interface gets slow and opening and viewing changes in each file is hard and painful.

Also please make it sure to check all your changes before a reviewer see them. The first file I saw (`.github/PULL_REQUEST_TEMPLATE.md`) has a change that is not appropriate (the three blank lines are intentional).

**Comment by kwankyu** [comment]
> Separate `` ``x, y`` `` to `` ``x``, ``y`` `` and remove excessive newlines.

Good idea.

**Comment by gmou3** [comment]
> I recommend you to split this PR. Due to the huge number of files that this PR touches, the github interface gets slow and opening and viewing changes in each file is hard and painful.
 
Ok, I'll see what I can do.

> Also please make it sure to check all your changes before a reviewer see them. The first file I saw (`.github/PULL_REQUEST_TEMPLATE.md`) has a change that is not appropriate (the three blank lines are intentional).

I did check most of them. I was under the impression that `>=3` blank lines are excessive even in `.md` or `.rst` files.

**Comment by gmou3** [comment]
I broke it up into 4 commits which you can check separately (I think it is faster now).

**Comment by kwankyu** [comment]
> I did check most of them.

OK.

> I was under the impression that `>=3` blank lines are excessive even in `.md` or `.rst` files.

I agree. Just that `.github/PULL_REQUEST_TEMPLATE.md` is special. The three spaces are for the author to fill the middle line.

> I broke it up into 4 commits which you can check separately (I think it is faster now).

Good idea! It works.

**Comment by kwankyu** [comment]
> I broke it up into 4 commits which you can check separately (I think it is faster now).

No.. I doesn't work. I am using "viewed" check boxes. For a commit, the boxes do not appear.

**Comment by gmou3** [comment]
So... separate PRs?

**Comment by kwankyu** [comment]
Strange. Some files have "check" boxes, but some other do not. 

Just leave it for now. Let me find a way...

File: src/sage/combinat/designs/difference_matrices.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def find_product_decomposition(g, k, lmbda=1):
 
     INPUT:
 
-    - ``g,k,lmbda`` -- integers, parameters of the difference matrix
+    - ``g``, ``k``, ``lmbda`` -- integers, parameters of the difference matrix
```
Comment (kwankyu) [inline src/sage/combinat/designs/difference_matrices.py:30]: `lmbda` seems to be a systematic typo throughout this method. It should be `lambda`. You may be acute on this misspelling :-)

It is also okay not to fix this if you will.

File: src/sage/combinat/designs/incidence_structures.py [hunk 2]
```diff
@@ -1486,10 +1486,10 @@ def is_t_design(self, t=None, v=None, k=None, l=None, return_parameters=False):
...
         INPUT:
 
-        - ``t,v,k,l`` (integers) -- their value is set to ``None`` by
-          default. The function tests whether the design is a ``t-(v,k,l)``
-          design using the provided values and guesses the others. Note that
-          `l`` cannot be specified if ``t`` is not.
+        - ``t``, ``v``, ``k``, ``l`` (integers) -- their value is set to
+          ``None`` by default. The function tests whether the design is a
+          ``t-(v,k,l)`` design using the provided values and guesses the
+          others. Note that ``l`` cannot be specified if ``t`` is not.
```
Comment (kwankyu) [inline src/sage/combinat/designs/incidence_structures.py:1492]: math
```
`t-(v,k,l)` 
```

File: src/sage/combinat/designs/orthogonal_arrays.py [hunk 3]
```diff
@@ -1879,7 +1878,7 @@ def OA_from_Vmt(m,t,V):
 
     INPUT:
 
-    - ``m,t`` (integers)
+    - ``m``, ``t`` (integers)
```
Comment (kwankyu) [inline src/sage/combinat/designs/orthogonal_arrays.py:1880]:
```
-- integers
```
?

File: src/sage/combinat/designs/orthogonal_arrays.py [hunk 4]
```diff
@@ -1930,7 +1929,7 @@ def QDM_from_Vmt(m,t,V):
...
```
Comment (kwankyu) [inline src/sage/combinat/designs/orthogonal_arrays.py:1931]: the same

Comment (kwankyu) [reply src/sage/combinat/designs/orthogonal_arrays.py:1931]: and many more below.

File: src/sage/combinat/designs/orthogonal_arrays_build_recursive.py [hunk 5]
```diff
@@ -189,9 +190,9 @@ def construction_3_5(k,n,m,r,s,t,explain_construction=False):
 
     INPUT:
 
-    - ``k,n,m`` (integers)
+    - ``k``, ``n``, ``m`` (integers)
```
Comment (kwankyu) [inline src/sage/combinat/designs/orthogonal_arrays_build_recursive.py:194]:
```
-- integers
```

Comment (kwankyu) [reply src/sage/combinat/designs/orthogonal_arrays_build_recursive.py:194]: and many more below

File: src/sage/combinat/matrices/latin.py [hunk 6]
```diff
@@ -793,8 +793,8 @@ def random_empty_cell(self):
 
         OUTPUT:
 
-        - ``[r, c]`` -- cell such that self[r, c] is empty, or returns
-          None if self is a (full) latin square.
+        - ``[r, c]`` -- cell such that ``self[r, c]`` is empty, or returns
+          ``None`` if ``self`` is a (full) latin square
```
Comment (kwankyu) [inline src/sage/combinat/matrices/latin.py:797]:
```
Latin square
```

File: src/sage/dynamics/arithmetic_dynamics/wehlerK3.py [hunk 7]
```diff
@@ -1723,8 +1723,9 @@ def lambda_minus(self, P, v, N, m, n, prec=100):
 
         - ``v`` -- non-negative integer. a place, use v = 0 for the Archimedean place
 
-        - ``m,n`` -- positive integers, We compute the local height for the divisor `E_{mn}^{+}`.
-                    These must be indices of non-zero coordinates of the point ``P``.
+        - ``m``, ``n`` -- positive integers; we compute the local height for
+          the divisor `E_{mn}^{+}`. These must be indices of non-zero
+          coordinates of the point ``P``.
 
         - ``prec`` -- (default: 100) float point or p-adic precision
```
Comment (kwankyu) [inline src/sage/dynamics/arithmetic_dynamics/wehlerK3.py:1730]: Another instance where plain 100 is perfectly okay :-)

File: src/sage/functions/orthogonal_polys.py [hunk 8]
```diff
@@ -2849,7 +2849,7 @@ class Func_meixner(OrthogonalFunction):
 
     - ``n`` -- the degree
     - ``x`` -- the independent variable `x`
-    - ``b, c`` -- the parameters `b, c`
+    - ``b``, ``c`` -- the parameters ``b``, ``c``
     """
```
Comment (kwankyu) [inline src/sage/functions/orthogonal_polys.py:2853]: math
```
`b`, `c`
```
?

File: src/sage/functions/orthogonal_polys.py [hunk 9]
```diff
@@ -2975,7 +2975,7 @@ class Func_hahn(OrthogonalFunction):
 
     - ``k`` -- the degree
     - ``x`` -- the independent variable `x`
-    - ``a, b`` -- the parameters `a, b`
+    - ``a``, ``b`` -- the parameters ``a``, ``b``
     - ``n`` -- the number of discrete points
```
Comment (kwankyu) [inline src/sage/functions/orthogonal_polys.py:2980]: here too.

File: src/sage/geometry/riemannian_manifolds/surface3d_generators.py [hunk 10]
```diff
@@ -110,7 +110,7 @@ def Dini(a=1, b=1, name="Dini's surface"):
 
         INPUT:
 
-        - ``a, b`` -- surface parameters.
+        - ``a``, ``b`` -- surface parameters.
```
Comment (kwankyu) [inline src/sage/geometry/riemannian_manifolds/surface3d_generators.py:113]: remove `.`?

File: src/sage/graphs/base/c_graph.pyx [hunk 11]
```diff
@@ -897,7 +897,7 @@ cdef class CGraph:
 
         INPUT:
 
-         - ``u, v`` -- non-negative integers, must be in self
+         - ``u``, ``v`` -- non-negative integers, must be in self
```
Comment (kwankyu) [inline src/sage/graphs/base/c_graph.pyx:901]:
```
``self``
```

Comment (kwankyu) [reply src/sage/graphs/base/c_graph.pyx:901]: many more below.

No need to fix if you will. Out of scope of this PR.

File: src/sage/graphs/base/dense_graph.pyx [hunk 12]
```diff
@@ -360,7 +360,7 @@ cdef class DenseGraph(CGraph):
 
         INPUT:
 
-        - ``u, v`` -- integers from 0, ..., n-1, where n is the number of vertices
+        - ``u``, ``v`` -- integers from 0, ..., n-1, where n is the number of vertices
             arc_labels -- must be a pointer to an (allocated) integer array
```
Comment (kwankyu) [inline src/sage/graphs/base/dense_graph.pyx]: math
```
`0`, ..., `n-1`
```
?

File: src/sage/combinat/matrices/latin.py (hunk 6 above)
Comment (gmou3) [reply src/sage/combinat/matrices/latin.py:797]: It is `latin` in many places. No need to capitalize I think.

File: src/sage/graphs/base/sparse_graph.pyx [hunk 13]
```diff
@@ -890,7 +890,7 @@ cdef class SparseGraph(CGraph):
 
         INPUT:
 
-         - ``u, v`` -- non-negative integers, must be in self
+         - ``u``, ``v`` -- non-negative integers, must be in self
```
Comment (kwankyu) [inline src/sage/graphs/base/sparse_graph.pyx:894]:
```
``self``
```

File: src/sage/graphs/base/sparse_graph.pyx [hunk 14]
```diff
@@ -924,7 +924,7 @@ cdef class SparseGraph(CGraph):
 
         INPUT:
 
-        - ``u, v`` -- integers from `0, ..., n-1`, where `n` is the number of vertices
+        - ``u``, ``v`` -- integers from `0, ..., n-1`, where `n` is the number of vertices
```
Comment (kwankyu) [inline src/sage/graphs/base/sparse_graph.pyx:929]:
```
`0`, ..., `n-1`
```
might look better. I am not sure though.

File: src/sage/graphs/base/sparse_graph.pyx [hunk 15]
```diff
@@ -954,7 +954,7 @@ cdef class SparseGraph(CGraph):
 
         INPUT:
 
-        - ``u, v`` -- integers from 0, ..., n-1, where n is the number of vertices
+        - ``u``, ``v`` -- integers from 0, ..., n-1, where n is the number of vertices
             arc_labels -- must be a pointer to an (allocated) integer array
```
Comment (kwankyu) [inline src/sage/graphs/base/sparse_graph.pyx]: math?

Comment (kwankyu) [reply src/sage/graphs/base/sparse_graph.pyx]: more below.

File: src/sage/graphs/generators/distance_regular.pyx [hunk 16]
```diff
@@ -1085,7 +1085,7 @@ def GrassmannGraph(const int q, const int n, const int input_e):
     INPUT:
 
     - ``q`` -- a prime power
-    - ``n, e`` -- integers with ``n > e+1``
+    - ``n``, ``e`` -- integers with ``n > e+1``
```
Comment (kwankyu) [inline src/sage/graphs/generators/distance_regular.pyx:1088]: math?

File: src/sage/graphs/strongly_regular_db.pyx [hunk 17]
```diff
@@ -2786,7 +2786,7 @@ def strongly_regular_graph(int v, int k, int l, int mu=-1, bint existence=False,
 
     INPUT:
 
-    - ``v,k,l,mu`` (integers) -- note that ``mu``, if unspecified, is
+    - ``v``, ``k``, ``l``, ``mu`` -- integers -- note that ``mu``, if unspecified, is
       automatically determined from ``v,k,l``.
```
Comment (kwankyu) [inline src/sage/graphs/strongly_regular_db.pyx:2791]:
```
``v``, ``k``, ``l``
```
?

File: src/sage/modular/modsym/heilbronn.pyx [hunk 18]
```diff
@@ -525,7 +525,7 @@ def hecke_images_gamma0_weight2(int u, int v, int N, indices, R):
     """
     INPUT:
 
-    - ``u, v, N`` -- integers so that gcd(u,v,N) = 1
+    - ``u``, ``v``, ``N`` -- integers so that gcd(u,v,N) = 1
     - ``indices`` -- a list of positive integers
```
Comment (kwankyu) [inline src/sage/modular/modsym/heilbronn.pyx:525]: math?

Comment (kwankyu) [reply src/sage/modular/modsym/heilbronn.pyx:525]: more below.

File: src/sage/modules/free_module.py [hunk 19]
```diff
@@ -3280,7 +3280,7 @@ def random_element(self, prob=1.0, *args, **kwds):
            probability `1-prob`. Otherwise coefficients will be chosen
            randomly from base ring (and may be zero).
 
-        - ``*args, **kwds`` -- passed on to ``random_element()`` function
+        - ``*args``, ``**kwds`` -- passed on to ``random_element()`` function
            of base ring.
```
Comment (kwankyu) [inline src/sage/modules/free_module.py]:
```
:func:`random_element`
```

File: src/sage/rings/polynomial/ore_function_field.py [hunk 20]
```diff
@@ -582,7 +582,7 @@ def random_element(self, degree=2, monic=False, *args, **kwds):
         - ``monic`` -- (default: ``False``) if ``True``, return a monic
           Ore function with monic numerator and denominator
 
-        - ``*args, **kwds`` -- passed in to the ``random_element`` method
+        - ``*args``, ``**kwds`` -- passed in to the ``random_element`` method
           for the base ring
```
Comment (kwankyu) [inline src/sage/rings/polynomial/ore_function_field.py]:
```
:meth:`random_element`
```

**Comment by kwankyu** [comment]
Otherwise lgtm.

File: src/sage/combinat/matrices/latin.py (hunk 6 above)
Comment (kwankyu) [reply src/sage/combinat/matrices/latin.py:797]: OK. Out of scope of this PR.

File: src/sage/geometry/riemannian_manifolds/surface3d_generators.py (hunk 10 above)
Comment (gmou3) [reply src/sage/geometry/riemannian_manifolds/surface3d_generators.py:113]: to be done in another PR

Comment (kwankyu) [reply src/sage/geometry/riemannian_manifolds/surface3d_generators.py:113]: OK.

File: src/sage/combinat/designs/orthogonal_arrays.py (hunk 3 above)
Comment (gmou3) [reply src/sage/combinat/designs/orthogonal_arrays.py:1880]: I'll leave these changes for another PR. There exist also `(boolean)` etc, and many type declarations are before the `--`.

File: src/sage/combinat/designs/difference_matrices.py (hunk 1 above)
Comment (gmou3) [reply src/sage/combinat/designs/difference_matrices.py:30]: Oh, trust me. Such things don't elude me.

**Comment by gmou3** [comment]
Best to check `src/sage/modular/modsym/heilbronn.pyx` when the documentation is ready.

File: src/sage/combinat/designs/difference_matrices.py (hunk 1 above)
Comment (gmou3) [reply src/sage/combinat/designs/difference_matrices.py:30]: I had to revert these changes because `lambda` is a reserved keyword.

Comment (kwankyu) [reply src/sage/combinat/designs/difference_matrices.py:30]: Right. That was why :-) 

Sorry about this.

File: src/sage_setup/autogen/interpreters/generator.py [hunk 21]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """Implements the generic interpreter generator."""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/generator.py]: Having a blank line seems better.

Comment (kwankyu) [reply src/sage_setup/autogen/interpreters/generator.py]: the same below

File: src/sage_setup/autogen/interpreters/instructions.py [hunk 22]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """Implements generic interpreter instructions and related utilities."""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/instructions.py]: Having a blank line seems better.

File: src/sage_setup/autogen/interpreters/memory.py [hunk 23]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """General purpose MemoryChunk types and related utilities"""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/memory.py]: Having a blank line seems better.

File: src/sage_setup/autogen/interpreters/specs/base.py [hunk 24]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """Base classes for interpreter specs."""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/specs/base.py]: Having a blank line seems better.

File: src/sage_setup/autogen/interpreters/storage.py [hunk 25]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """Implements different data storage types."""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/storage.py]: Having a blank line seems better.

File: src/sage_setup/autogen/interpreters/utils.py [hunk 26]
```diff
@@ -8,7 +8,6 @@
 # (at your option) any later version.
 #                  http://www.gnu.org/licenses/
 #*****************************************************************************
-
 """Miscellaneous utility routines used for the interpreter generator."""
```
Comment (kwankyu) [inline src/sage_setup/autogen/interpreters/utils.py]: Having a blank line seems better.

File: src/sage/modular/modsym/heilbronn.pyx [hunk 27]
```diff
@@ -229,20 +226,20 @@ cdef class Heilbronn:
...
 
     def apply(self, int u, int v, int N):
-        """
-        Return a list of pairs ((c,d),m), which is obtained as follows:
+        r"""
+        Return a list of pairs `((c,d),m)`, which is obtained as follows:
 
-        1) Compute the images (a,b) of the vector (u,v) (mod N) acted on by
+        1) Compute the images `(a,b)` of the vector `(u,v) (mod N)` acted on by
         each of the HeilbronnCremona matrices in self.
```
Comment (kwankyu) [inline src/sage/modular/modsym/heilbronn.pyx:233]:
```
\mod
```
?

Comment (gmou3) [reply src/sage/modular/modsym/heilbronn.pyx:233]: \pmod

Comment (kwankyu) [reply src/sage/modular/modsym/heilbronn.pyx:233]: Yes.

**Review by kwankyu — APPROVED** [review]
Thanks!

**Pull request merged**